src/components/image-uploader/image-uploader.js

```diff
@@ -3,80 +3,147 @@ import { Button, Loader, Segment, Icon } from "semantic-ui-react";
 
 import "./image-uploader.css";
 import previewImage from "../../images/preview-image.svg";
-import image from "../../images/image.png";
 
-export const imageUploader = (props) => {
-	return (
-		<div className="image-uploader">
-			<div className="image-uploader__container">
-				<div className="image-uploader__header">
-					{props.status === "upload" && (
-						<>
-							<div className="image-uploader__title">Upload Your Image</div>
-							<div className="image-uploader__description">
-								File should be jpeg, png,...
+export class ImageUploader extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.uploadInput = React.createRef();
+		this.focusUploadInputHandler = this.focusUploadInputHandler.bind(this);
+		this.fileUploadHanlder = this.fileUploadHanlder.bind(this);
+		this.onChangeFileHandler = this.onChangeFileHandler.bind(this);
+		this.onDropHandler = this.onDropHandler.bind(this);
+		this.onDragOverHandler = this.onDragOverHandler.bind(this);
+		this.copyLinkHandler = this.copyLinkHandler.bind(this);
+	}
+
+	fileUploadHanlder(file, status) {
+		if (this.props.onChangeHandler) {
+			this.props.onChangeHandler(file, status);
+		}
+	}
+
+	onChangeFileHandler(e) {
+		if (e.target.files && e.target.files.length > 0) {
+			this.fileUploadHanlder(e.target.files, "success");
+		}
+	}
+
+	focusUploadInputHandler() {
+		this.uploadInput.current.click();
+	}
+
+	onDragOverHandler(e) {
+		e.preventDefault();
+	}
+
+	onDropHandler(e) {
+		e.preventDefault();
+		this.fileUploadHanlder(e.dataTransfer.files, "success");
+	}
+
+	copyLinkHandler() {
+		navigator.clipboard
+			.writeText(this.props.link)
+			.then(() => {
+				alert("Copied link!!!");
+			})
+			.catch((err) => {
+				console.log(err);
+			});
+	}
+
+	render() {
```

Comment:
render function của component này to quá, a có thể thử chia thành 3 components nhỏ giống như trong design và để cái app render cái component cần thiết?